drivers/ethernet/eth_stm32_hal.c

```diff
@@ -773,6 +775,103 @@ static int eth_initialize(const struct device *dev)
 	return 0;
 }
 
+static uint32_t reverse(uint32_t val)
+{
+	uint32_t res = 0;
+	int i;
+
+	for (i = 0; i < 32; i++) {
+		if (val & (1 << i)) {
+			res |= 1 << (31 - i);
+		}
+	}
+
+	return res;
+}
+
+static void net_if_mcast_cb(struct net_if *iface,
+			    const struct net_addr *addr,
+			    bool is_joined)
+{
+	ARG_UNUSED(addr);
+	ARG_UNUSED(is_joined);
+
+	const struct device *dev;
+	struct eth_stm32_hal_dev_data *dev_data;
+	ETH_HandleTypeDef *heth;
+#if defined(CONFIG_NET_NATIVE_IPV6)
+	struct net_if_ipv6 *ipv6;
+#endif
+#if defined(CONFIG_NET_NATIVE_IPV4)
+	struct net_if_ipv4 *ipv4;
+#endif
+	struct net_eth_addr mac_addr;
+	uint32_t crc;
+	uint32_t hash_table[2];
+	uint32_t hash_index;
+	int i;
+
+	__ASSERT_NO_MSG(iface != NULL);
+
+	dev = net_if_get_device(iface);
+	__ASSERT_NO_MSG(dev != NULL);
+
+	dev_data = DEV_DATA(dev);
+	__ASSERT_NO_MSG(dev_data != NULL);
+
+	heth = &dev_data->heth;
+	__ASSERT_NO_MSG(heth != NULL);
+
+	hash_table[0] = 0;
+	hash_table[1] = 0;
+
+#if defined(CONFIG_NET_NATIVE_IPV6)
+	if (net_if_config_ipv6_get(iface, &ipv6) < 0) {
+		return;
+	}
+
+	for (i = 0; i < NET_IF_MAX_IPV6_MADDR; i++) {
+		if (!ipv6->mcast[i].is_used) {
+			continue;
+		}
+
+		net_eth_ipv6_mcast_to_mac_addr(&ipv6->mcast[i].address.in_addr,
+					       &mac_addr);
+		crc = reverse(crc32_ieee(mac_addr.addr,
+					 sizeof(struct net_eth_addr)));
+		hash_index = (crc >> 26) & 0x3f;
+		hash_table[hash_index / 32] |= (1 << (hash_index % 32));
```

Comment:
```suggestion
		hash_table[hash_index % 32] |= (1 << (hash_index % 32));
```

Wouldn't be i more correct?